### PR TITLE
Improve Chain/Transport unit test coverage  - Closes #4595

### LINKED ISSUE #4595: Add missing unit tests coverage for Chain/Transport banning mechanism
### Description

- Improve test coverage in `Chain/Transport` related to LIP-0004 Peer banning mechanism.

### Acceptance Criteria

- Have unit test coverage in `Chain/Transport` for every Peer banning scenario.

### PR DESCRIPTION
**Pull request by valamidev**
### What was the problem?
- Unit test were missing in some edge cases related to `LIP-0004` Peer banning mechanism.

### How did I solve it?
- Add missing test cases where schema validator fail should apply penalty.

### How to manually test it?
- `npm run mocha:unit -- transport`

### Review checklist

- [x] The PR resolves #4595 
- [x] All new code is covered with unit tests
- [x] Relevant integration / functional tests are added
- [x] Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
- [x] Documentation has been added/updated

### CODE REVIEW

**Comment by valamidev** [comment]
> Please, scan the file again. I count 10 cases of invoking `network:applyPenalty` action in transport. However, only two are addressed in the PR.

Those are covered in [jest/unit/specs/modules/chain/transport/transport.spec.js](https://github.com/LiskHQ/lisk-sdk/blob/4595-unit-test-improvement-chain-transport/framework/test/jest/unit/specs/modules/chain/transport/transport.spec.js)

File: framework/test/mocha/unit/modules/chain/transport/transport.js [hunk 1]
```diff
@@ -614,13 +612,24 @@ describe('transport', () => {
...
 								expect(e[0].message).to.equal(
 									"should have required property 'blockId'",
 								);
+								expect(channelStub.invoke).toHaveBeenCalledWith(
+									'network:applyPenalty',
+									{
+										peerId: defaultPeerId,
+										penalty: 100,
+									},
+								);
```
[the remark this answers is not in the document]
Comment (valamidev) [reply framework/test/mocha/unit/modules/chain/transport/transport.js]: Awesome catch!
